ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto

```diff
@@ -188,6 +189,32 @@ message ExerciseByKeyCommand {
   Value choice_argument = 4;
 }
 
+// Exercise a choice on an existing contract specified by its key.
+message ExerciseByInterfaceKeyCommand {
+  // The interface id that defines the key.
+  // Required
+  Identifier key_interface_id = 1;
```

Comment:
Can this be a template, so that we also close the existing gap of being unable to exerciseByKey when choice is on an interface

Reply:
not planned at the moment. I'm also not sure it would really be very useful. I expect most models will either use contract keys or interface keys but not both.